speech/cloud-client/src/main/java/com/example/speech/Recognize.java

```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
     }
   }
   // [END speech_transcribe_file_with_metadata]
+
+  // [START speech_transcribe_diarization]
+  /**
+   * Transcribe the given audio file using speaker diarization.
+   *
+   * @param fileName the path to an audio file.
+   */
+  public static void transcribeDiarization(String fileName) throws Exception {
+    Path path = Paths.get(fileName);
+    byte[] content = Files.readAllBytes(path);
+
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      // Get the contents of the local audio file
+      RecognitionAudio recognitionAudio =
+          RecognitionAudio.newBuilder().setContent(ByteString.copyFrom(content)).build();
+
+      // Configure request to enable enhanced models
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(8000)
+              .setEnableSpeakerDiarization(true)
+              .setDiarizationSpeakerCount(2)
+              .setEnableAutomaticPunctuation(true)
+              .setModel("phone_call")
+              .build();
+
+      // Perform the transcription request
+      RecognizeResponse recognizeResponse = speechClient.recognize(config, recognitionAudio);
+
+      // Print out the results
+      for (SpeechRecognitionResult result : recognizeResponse.getResultsList()) {
+        // There can be several alternative transcripts for a given chunk of speech. Just
+        // use the first (most likely) one here.
+        SpeechRecognitionAlternative alternative = result.getAlternatives(0);
+        System.out.format("Transcript : %s\n", alternative.getTranscript());
+        System.out.format(
+            "Speaker Tag : %s \n",
+            alternative.getWords((alternative.getWordsCount() - 1)).getSpeakerTag());
+        System.out.format(
+            "Word: %s\n\n", alternative.getWords((alternative.getWordsCount() - 1)).getWord());
```

Comment:
Lines: 838-841:
Is this printing out the last speaker and their last word?

Is is possible to maybe do something like: (Not sure what the results look like)
Speaker Tag ###: Hey, how are you?
Speaker Tag ***: I'm doing good. 

Thoughts?

Reply:
It is printing out the last speaker and last word. The words array contains the entire transcript up until that point. 
Speaker Tag ###: Hey, how are you? : Definitely makes more sense. Will switch it out.

Reply:
Can you add this as a comment? ie the explanation of why you're getting the last word of the alternative instead of, say, all the words or the first word.